MeshLib/MeshGenerators/MeshLayerMapper.cpp

```diff
@@ -60,6 +61,10 @@ MeshLib::Mesh* MeshLayerMapper::createStaticLayers(MeshLib::Mesh const& mesh, st
 	std::vector<MeshLib::Node*> new_nodes(nNodes + (nLayers * nNodes));
 	std::vector<MeshLib::Element*> new_elems;
 	new_elems.reserve(nElems * nLayers);
+	MeshLib::Properties properties;
+	boost::optional<PropertyVector<int> &> materials = 
+		properties.createNewPropertyVector<int>("MaterialIDs", MeshLib::MeshItemType::Cell);
+	materials->reserve(nElems * nLayers);
```

Comment:
If the `materials` after the `createNewPropertyVector` call has an uninitialized optional object, than `*materials` contains no valid value and a nullptr in best case. Calling `reserve` on an arbitrary pointer usually leads to a segv.
There must be a check
```c++
if (!materials)
{
    ERR("Could not create bla bla");
    std::abort();
}
```
then one is free to call `materials->reserve(...)`.